File: bench/go/matrix.go
package main

import (
	"fmt"
	"time"
)

func main() {
	N := 200
	A := make([]float64, N*N)
	B := make([]float64, N*N)
	C := make([]float64, N*N)

	for i := 0; i < N*N; i++ {
		A[i] = float64(i % 100)
		B[i] = float64((i * 7) % 100)
	}

	start := time.Now()

	// ikj loop order for cache efficiency (matches C baseline)
	for i := 0; i < N; i++ {
		for k := 0; k < N; k++ {
			aik := A[i*N+k]
			for j := 0; j < N; j++ {
				C[i*N+j] += aik * B[k*N+j]
			}
		}
	}

	elapsed := time.Since(start).Seconds() * 1000

	fmt.Printf("C[0][0]=%.0f C[%d][%d]=%.0f\n", C[0], N-1, N-1, C[(N-1)*N+(N-1)])
	fmt.Printf("RESULT:%.3f\n", elapsed)
}

File: bench/python/matrix.py
import time

N = 500

A = [0.0] * (N * N)
B = [0.0] * (N * N)
C = [0.0] * (N * N)

for i in range(N * N):
    A[i] = float(i % 100)
    B[i] = float((i * 7) % 100)

start = time.perf_counter()

# ikj loop order for cache efficiency (matches C baseline)
for i in range(N):
    for k in range(N):
        a_ik = A[i * N + k]
        for j in range(N):
            C[i * N + j] += a_ik * B[k * N + j]

end = time.perf_counter()
ms = (end - start) * 1000

print(f"C[0][0]={C[0]:.0f} C[499][499]={C[499 * N + 499]:.0f}")
print(f"RESULT:{ms:.3f}")
